[CI] PreallocateTests testPreallocate failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3416 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/kixgradviemo2)
- [elasticsearch-periodic-platform-support #3416 / windows-2019_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/4t7rwqs2wqlp2)
- [elasticsearch-periodic-platform-support #3416 / windows-2016_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/5cg636fjc2eeu)
- [elasticsearch-periodic #3442 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/siov7gxnujcja)
- [elasticsearch-pull-request #26046 / part-1-windows](https://gradle-enterprise.elastic.co/s/dj72kiwwkddkg)

**Reproduction Line:**
```
gradlew ":libs:elasticsearch-native:test" --tests "org.elasticsearch.nativeaccess.PreallocateTests.testPreallocate" -Dtests.seed=2CBD5DEA907DA216 -Dtests.locale=sr-BA -Dtests.timezone=America/Vancouver -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testPreallocate'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.nativeaccess.PreallocateTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <1048576L>
     but: was <0L>
```

**Issue Reasons:**
- [main] 5 failures in test testPreallocate (17.9% fail rate in 28 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/libs/preallocate/src/test/java/org/elasticsearch/preallocate/PreallocateTests.java
+++ b/libs/preallocate/src/test/java/org/elasticsearch/preallocate/PreallocateTests.java
@@ -18,13 +18,12 @@ import static org.hamcrest.Matchers.equalTo;
 
 public class PreallocateTests extends ESTestCase {
     public void testPreallocate() throws Exception {
+        assumeFalse("no preallocate on windows", System.getProperty("os.name").startsWith("Windows"));
         Path cacheFile = createTempFile();
         long size = 1024 * 1024; // 1 MB
         Preallocate.preallocate(cacheFile, size);
         OptionalLong foundSize = FileSystemNatives.allocatedSizeInBytes(cacheFile);
         assertTrue(foundSize.isPresent());
-        // Note that on Windows the fallback setLength is used. Although that creates a sparse
-        // file on Linux/MacOS, it full allocates the file on Windows
         assertThat(foundSize.getAsLong(), equalTo(size));
     }
 }

